### PR TITLE
Add a test for GetGroup never returning EmptyGroup

### LINKED ISSUE #914: Write a test for GetGroup never returning Empty Group
We merged #253 without a corresponding test.  We need to write one.

### PR DESCRIPTION
**Pull request by gucluoz**
This will test previously fixed #253 via public API surface.

Also fix #914

### CODE REVIEW

**Comment by stephentoub** [comment]
cc; @Priya91, @ellismg

File: src/System.Text.RegularExpressions/tests/GroupCollectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        Match matchPos = regex.Match(inputWithGroup);
+        Match matchNeg = regex.Match(inputWithoutGroup);
+
+        //test if Success returns true with group(s) present
+        Assert.True(matchPos.Groups[0].Success);
+
+        //public [] operator calling internal GetGroup()
+        //should return internal _emptygroup thus making
+        //call to Success() false
+        Assert.False(matchNeg.Groups[0].Success);
```
Comment (hughbe) [inline src/System.Text.RegularExpressions/tests/GroupCollectionTests.cs]: Nice addition. I was just thinking that a similar test should be worked into #7046 once that PR is merged.

In [this](https://github.com/hughbe/corefx/blob/regex-test-cleanup-2/src/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs#L301) method we can add an assertion
`Assert.Equal(expectedSuccess, match.Groups[0].Success)`

I'm just saying this here 'cause its likely this PR will be merged before mine (mine's pretty big), so there's an example of a future contribution

File: src/System.Text.RegularExpressions/tests/GroupCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using Xunit;
+
+public class GroupCollectionTests
```
Comment (hughbe) [inline src/System.Text.RegularExpressions/tests/GroupCollectionTests.cs]: Nit: missing a namespace (`System.Text.RegularExpressions.Tests`)

**Comment by Priya91** [comment]
LGTM. Thanks for the contribution.

**Pull request merged**